### DISCUSSION TITLE
Prefect Cloud can only schedule 10 runs in advance, but I need more.

### DISCUSSION THREAD

**Discussion by TilakMaddy** [PrefectHQ/prefect#5177] [Q&A]
I'd like to create an ETL pipeline flow which is Parametric, so each time a user on our website wants to schedule a flow from a source to destination we just substitute the parameters and run it. Now obviously we have a lot of flows that need to be scheduled but from the same template

I get the following warning when I scroll through the calendar past 10 days. Such a warning has not been documented in the website.
<img width="1280" alt="prefect_issue" src="https://user-images.githubusercontent.com/11134428/143864002-4bf75d91-7bdc-4711-8482-2b209df2ed9f.png">

The alt. is to register a new flow for every user (not practical).

- Is it because I am restricted by the starter plan or is there such a small limit for everyone. Since I work at a startup I want to know how many maximum flows can we afford on Prefect Cloud with other bigger plans ?

- As of now we are not interested in hosting our own prefect cloud but I'd like to know what real advantages do we have for hosting our own prefect server ? I think prefect cloud is super cool !

- Since we are going to have a lot of data we would need history of  more than just 2 weeks. So what do I do ?

Thanks 🙏

**Comment by TilakMaddy** [comment]
ANY HELP is appreciated !

**Reply by zanieb** [reply]
Hi. If your flow is run on a schedule (i.e. daily), then only 10 runs will be scheduled at a time. Each time they run, we'll schedule more runs for you. If you want to schedule a flows yourself (for example, based on user input), you can schedule as many as you'd like. There is no limit.

- For a Server/Cloud comparison see https://docs.prefect.io/orchestration/server/overview.html#prefect-server-vs-prefect-cloud-which-should-i-choose
- For additional history retention, you can contact sales@prefect.io
